src/cargo/util/progress.rs

```diff
@@ -76,6 +77,76 @@ struct Format {
     max_print: usize,
 }
 
+/// Taskbar progressbar
+///
+/// Outputs ANSI codes according to the `Operating system commands`.
+/// Currently only works in the Windows Terminal and ConEmu.
+struct TaskbarProgress {
+    show: bool,
+    pub percent: u8,
+}
+
+impl TaskbarProgress {
+    /// Creates a new `TaskbarProgress` from a cargo's config system.
+    ///
+    /// * `config == None` enables taskbar progress reporting on Windows
+    /// by default and disables on other platforms
+    fn from_config(config: Option<bool>) -> Self {
+        let show = match config {
+            Some(v) => v,
+            None => cfg!(target_os = "windows"),
+        };
+
+        TaskbarProgress { show, percent: 0 }
+    }
+
+    /// Removes a progress bar from taskbar
+    pub fn remove(&self) -> TaskbarValue {
+        if self.show {
+            TaskbarValue::Remove
+        } else {
+            TaskbarValue::None
+        }
+    }
+
+    /// Shows a value from a `percent` field
+    pub fn value(&self) -> TaskbarValue {
+        if self.show {
+            TaskbarValue::Value(self.percent)
+        } else {
+            TaskbarValue::None
+        }
+    }
+}
+
+/// A taskbar progress value printable as ANSI OSC escape code
+enum TaskbarValue {
+    /// Do not output anything
+    None,
+    /// Remove progress
+    Remove,
+    /// Progress value 0-100
+    Value(u8),
+}
+
+impl std::fmt::Display for TaskbarValue {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        // From https://conemu.github.io/en/AnsiEscapeCodes.html#ConEmu_specific_OSC
+        // ESC ] 9 ; 4 ; st ; pr ST
+        // When st is 0: remove progress.
+        // When st is 1: set progress value to pr (number, 0-100).
+        // When st is 2: set error state in taskbar, pr is optional.
+        // When st is 3: set indeterminate state.
```

Comment:
Should we respect `ProgressStyle::Indeterminate` and have a way of outputting this?

Reply:
Good catch! We should, and it works beatifully:

https://github.com/user-attachments/assets/70ec0a05-3565-412c-be2e-fcf525522f99